dds/DCPS/NetworkConfigModifier.cpp

```diff
@@ -0,0 +1,232 @@
+        reinterpret_cast<sockaddr_in *> (p_if->ifa_addr);
+
+      // Sometimes the kernel returns 0.0.0.0 as the interface
+      // address, skip those...
+      if (addr->sin_addr.s_addr != INADDR_ANY) {
+        address.set((u_short) 0,
+                     addr->sin_addr.s_addr,
+                     0);
+
+        NetworkInterface iface(count, p_if->ifa_name, p_if->ifa_flags & IFF_MULTICAST);
+        iface.addresses.insert(address);
+
+        NetworkConfigMonitor::add_interface(iface);
+        NetworkConfigMonitor::add_address(count, address);
+
+        ++count;
+      }
+    }
+  }
+
+  ::freeifaddrs (p_ifa);
+
+  return true;
+}
+
+bool NetworkConfigModifier::close()
+{
+  if (DCPS_debug_level > 0) {
+    ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) NetworkConfigModifier::close()\n")));
+  }
+
+  return true;
+}
+
+void NetworkConfigModifier::add_interface(const OPENDDS_STRING &name)
+{
+  NetworkInterface* p_nic = 0;
+
+  ifaddrs* p_ifa = 0;
+  ifaddrs* p_if = 0;
+
+  ACE_INET_Addr address;
+
+  if (::getifaddrs(&p_ifa) != 0) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::add_interface %p"),
+        ACE_TEXT("getifaddrs error - ")));
+      return;
+  }
+
+  int count = 0;
+
+  // Pull the address out of each INET interface.
+  for (p_if = p_ifa; p_if != 0; p_if = p_if->ifa_next)
+  {
+    if (p_if->ifa_addr == 0)
+      continue;
+
+    // Check to see if it's up.
+    if ((p_if->ifa_flags & IFF_UP) != IFF_UP)
+      continue;
+
+    if (p_if->ifa_addr->sa_family == AF_INET) {
+      if (name == p_if->ifa_name) {
+        p_nic = new NetworkInterface (count, p_if->ifa_name, p_if->ifa_flags & IFF_MULTICAST);
+        break;
+      }
+
+      ++count;
+    }
+  }
+
+  ::freeifaddrs (p_ifa);
+
+  if (p_nic) {
+    NetworkConfigMonitor::add_interface(*p_nic);
+    delete p_nic;
+  }
+
+  validate_interfaces_index();
+}
+
+void NetworkConfigModifier::remove_interface(const OPENDDS_STRING &name)
+{
+  int index = get_index(name);
+  if (index != -1) {
+    NetworkConfigMonitor::remove_interface(index);
+  }
+  else {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::remove_interface. Interface '%s' not found"), name.c_str()));
+  }
+}
+
+void NetworkConfigModifier::add_address(const OPENDDS_STRING &name, const ACE_INET_Addr& address)
+{
+  int index = get_index(name);
+  if (index != -1) {
+    NetworkConfigMonitor::add_address(index, address);
+  }
+  else {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::add_address. Interface '%s' not found"), name.c_str()));
+  }
+}
+
+void NetworkConfigModifier::remove_address(const OPENDDS_STRING &name, const ACE_INET_Addr& address)
+{
+  int index = get_index(name);
+  if (index != -1) {
+    NetworkConfigMonitor::remove_address(index, address);
+  }
+  else {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::remove_address. Interface '%s' not found"), name.c_str()));
```

Comment:
```suggestion
    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::remove_address. Interface '%C' not found"), name.c_str()));
```

Reply:
This also apply to to other errors in set of operations. Also this shouldn't be shown unless debug level is active because it is a error without an error return to the user.